sound/soc/sof/intel/hda.c

```diff
@@ -1391,6 +1400,8 @@ static struct snd_soc_acpi_mach *hda_sdw_machine_select(struct snd_sof_dev *sdev
 		link_index = hweight32(link_mask & (BIT(peripherals->array[i]->bus->link_id) - 1));
 		links[link_index].adr_d = find_acpi_adr_device(sdev->dev, peripherals->array[i],
 							       &links[link_index], &amp_index);
+		if (!links[link_index].adr_d)
+			return NULL;
```

Comment:
This is an error? Should we print a message?

Reply:
Yes, this is an error, and the error message is printed in find_acpi_adr_device().